hicommonwealth/commonwealth
# Remove legacy email login feature.

## Link to Issue
Closes: #7864 
Closes: #6752 

## Description of Changes
High level: removes any form of email-based login (maintaining the Magic-based path, but using address alone), besides ghost address claiming. List of changes:
- Renamed `LoginTokens` to `EmailUpdateTokens` and truncated table to remove orphaned data.
- Ghost address claiming now takes place entirely within the magic passport route, no longer utilizing client-side logic to determine which addresses get claimed.
- Emails re-added to SSO derived information, as they are no longer considered high-risk login data (although they are still secure from a read perspective).
- Several routes removed or renamed:
  - /finishEmailLogin => /finishUpdateEmail; modified to no longer perform a _login_, just an update in the database following the verification
  - /startEmailLogin REMOVED
  - /updateAddress REMOVED

## Test Plan
- **Ensure all Magic-based login methods have not regressed.**
- Testing ghost addresses (NOTE: _Cosmos_ ghost address claiming was NOT considered and may be UNSUPPORTED):
  1. select a user id from the command `SELECT DISTINCT U.id as userid, U.email, A.community_id, T.id as threadid FROM "Addresses" A JOIN "Users" U ON A.user_id = U.id JOIN "Threads" T ON A.id = T.address_id WHERE A.ghost_address = true AND A.community_id = '1inch' AND T.community_id = '1inch';`
  2. update the user's email in the database to a UNIQUE email (not used for any other address) that you control, e.g. test@common.xyz (if you need to create a new email for claiming purposes, append a + symbol and then whatever to your common.xyz email, e.g. test+1234@common.xyz)
  3. sign in with test@common.xyz using the magic EMAIL flow (NOT SSO / GOOGLE, other SSO auth should NOT succeed at claiming the account)
  4. ensure that you land on the account and that the ghost address was deleted
- Testing email updating:
  - ensure the new user onboarding flow succeeds (may need some tweaking cc @mzparacha)
  - in "notification settings", ensure you can update an email (check spam for the validation email)
